docs/configuration.asciidoc

```diff
@@ -0,0 +1,265 @@
+[[configuration]]
+== Configuration
+
+[float]
+=== Gradle configuration
+
+Configure your application at compile time within your application's `build.gradle` file:
+
+[source,groovy]
+----
+// Android app's build.gradle file
+plugins {
+    //...
+    id "co.elastic.apm.android" version "[latest_version]"
+}
+
+elasticApm {
+    // Minimal configuration
+    serverUrl = "https://your.elastic.server"
+
+    // Optional
+    serviceName = "your app name" <1>
+    serviceVersion = "0.0.0" <2>
+    secretToken = "your server auth token" <3>
+}
+----
+<1> Defaults to your `android.defaultConfig.applicationId` value.
+<2> Defaults to your `android.defaultConfig.versionName` value.
+<3> Defaults to null.
+
+All of the values provided in the Gradle configuration can be overridden with the following environment variables:
+
+|===
+|Config |Associated Environment variable
+
+|serviceName
+|`ELASTIC_APM_SERVICE_NAME`
+
+|serviceVersion
+|`ELASTIC_APM_SERVICE_VERSION`
+
+|serverUrl
+|`ELASTIC_APM_SERVER_URL`
+
+|secretToken
+|`ELASTIC_APM_SECRET_TOKEN`
+
+|===
+
+[float]
+=== Runtime configuration
+
+The runtime configuration is provided within your https://developer.android.com/reference/android/app/Application[Application] class at the time of initializing the Elastic agent.
+This configuration overrides any previously-set compile time configuration.
+
+The way it works is by providing your own instance of the `ElasticApmConfiguration` class as shown below
+
+[source,java]
+----
+// Application class
+
+class MyApp extends android.app.Application {
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        ElasticApmAgent.initialize(this, ElasticApmConfiguration.builder().build());
+    }
+}
+----
+
+[float]
+==== APM server connectivity
+
+The server connectivity parameters can be provided at compile time, either by using the gradle dsl configuration, or by providing the server connectivity related environment variables as mentioned above.
+Later on, when the app is running, the connectivity parameters can be overriden by providing a custom `Connectivity` instance when initializing the Elastic agent.
+
+The `Connectivity` interface provides a set of convenient methods to avoid having to create your own implementation in case you need to define a custom connectivity.
```

Comment:
I'm not sure I understand what this sentence is trying to say.

Reply:
I see, that's interesting. I think this might be the case of adding extra information that might cause confusion instead of helping much 😅 - So basically, `Connectivity` is an interface, so anyone can just create their own implementation and that would do. However, there are some common use cases for which I created helper methods, which might be easier to use rather than creating your own class that implements `Connectivity`. You can see those methods [here](https://github.com/elastic/apm-agent-android/pull/46/files#diff-6817141da84f46b7b26749d24c6bab4f40ace13bbc872a001df621335810613c).

I was thinking about maybe adding some examples here, but it really isn't necessary to use said helper methods to make it work, so I guess it'd be better if we just added a link to the `Connectivity` interface Javadoc, in case people are interested in seeing the code doc for it, where all the helper methods are described. It might be better as well for maintainability in the future, in case this interface changes, then this doc won't get inaccurate. However, for now I don't have the Javadoc link since the library isn't deployed yet, so for now I'm just going to remove it to avoid confusion 👍